driver-core/src/test/java/com/datastax/driver/core/querybuilder/QueryBuilderExecutionTest.java

```diff
@@ -924,4 +927,45 @@ public void should_support_group_by() throws Exception {
                         .groupBy("b")))
         .containsExactly(row(1, 1, 12));
   }
+
+  /**
+   * Validates that {@link QueryBuilder} can construct a SELECT query for a materialized view.
+   *
+   * @test_category queries:builder
+   * @jira_ticket JAVA-2123
+   * @since 3.7.0
+   */
+  @CassandraVersion(
+      value = "3.0",
+      description = "Support for materialized views was added to C* 3.0")
+  @Test(groups = "short")
+  public void should_select_from_materialized_view() {
+
+    String table = TestUtils.generateIdentifier("table");
+    final String mv = TestUtils.generateIdentifier("mv");
+
+    execute(
+        String.format("CREATE TABLE %s (pk int, cc int, v int, PRIMARY KEY (pk, cc))", table),
+        String.format("INSERT INTO %s (pk, cc, v) VALUES (0,0,0)", table),
+        String.format("INSERT INTO %s (pk, cc, v) VALUES (0,1,1)", table),
+        String.format("INSERT INTO %s (pk, cc, v) VALUES (0,2,2)", table),
+        String.format(
+            "CREATE MATERIALIZED VIEW %s AS SELECT cc FROM %s WHERE cc IS NOT NULL PRIMARY KEY (pk, cc)",
+            mv, table));
+
+    // Wait until the MV is fully constructed
+    ConditionChecker.check()
+        .that(
+            new Callable<Boolean>() {
+              @Override
+              public Boolean call() {
+                return session().execute("SELECT * FROM " + mv).all().size() == 3;
+              }
+            })
+        .before(1, MINUTES)
+        .becomesTrue();
+
+    assertThat(session().execute(select().column("cc").as("mycc").from(mv)))
+        .containsExactly(row(0), row(1), row(2));
```

Comment:
this `row(n)` syntax is validating only the fact that row exists? or also the content of a row?

Reply:
It also asserts the contents, see `ResultSetAssert`.